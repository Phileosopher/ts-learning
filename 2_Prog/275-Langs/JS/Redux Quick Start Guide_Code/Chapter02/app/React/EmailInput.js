import React, { Component } from "react";

class EmailInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ""
    };
    this.handleEmailChange = this.handleEmailChange.bind(this);
  }

  handleEmailChange(event) {
    this.setState({ value: event.target.value });
  }

  render() {
    return (
      <div>
        <label>Enter Your Email</label>
        <input
          type="email"
          onChange={this.handleEmailChange}
          value={this.state.value}
          name="email"
        />
      </div>
    );
  }
}

export default EmailInput;
