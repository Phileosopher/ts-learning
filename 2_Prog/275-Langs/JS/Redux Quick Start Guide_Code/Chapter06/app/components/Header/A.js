import styled from 'styled-components';
import { Link } from 'react-router-dom';

const A = styled(Link)`
  padding: 5px 15px;
  display: inline-block;
`;

export default A;
