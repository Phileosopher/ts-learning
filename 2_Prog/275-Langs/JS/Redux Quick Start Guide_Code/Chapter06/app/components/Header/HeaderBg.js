import styled from 'styled-components';
import BG from './bgheader.png';

export default styled.div`
  min-height: 100px;
  background-image: url(${BG});
  background-color: #def0fb;
  background-position: 100% 0;
`;
