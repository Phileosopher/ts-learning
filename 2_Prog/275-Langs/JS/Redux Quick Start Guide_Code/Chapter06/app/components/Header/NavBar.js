import styled from 'styled-components';

export default styled.div`
  display: flex;
  padding-right: 10px;
  align-items: center;
  background-color: #204666;

  & > div {
    flex-shrink: 0;
    display: flex;
    align-items: center;

    &:last-child {
      flex: 1;
      justify-content: flex-end;
    }
  }

  button {
    line-height: 0;
  }
`;
