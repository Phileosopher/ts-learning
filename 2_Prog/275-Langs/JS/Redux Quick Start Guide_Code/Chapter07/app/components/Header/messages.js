import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.components.Header';

export default defineMessages({
  home: {
    id: `${scope}.home`,
    defaultMessage: 'Home',
  },
  features: {
    id: `${scope}.features`,
    defaultMessage: 'Features',
  },
  login: {
    id: `${scope}.login`,
    defaultMessage: 'Login',
  },
  register: {
    id: `${scope}.register`,
    defaultMessage: 'Register',
  },
  helloHeader: {
    id: `${scope}.hello`,
    defaultMessage: 'Hello',
  },
});
