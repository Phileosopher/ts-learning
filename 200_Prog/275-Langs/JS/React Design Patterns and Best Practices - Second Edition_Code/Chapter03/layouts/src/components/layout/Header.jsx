import React from 'react';
import { string } from 'prop-types';

const Header = ({ title }) => (
  <header className="App-header">
    <h1>{title}</h1>
  </header>
);

Header.propTypes = {
  title: string.isRequired
};

export default Header;
