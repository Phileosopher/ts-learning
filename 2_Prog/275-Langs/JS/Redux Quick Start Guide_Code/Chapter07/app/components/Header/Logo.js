import React from 'react';
import LogoImg from './logo.png';

export default props => (
  <img alt="" className="logo-header" src={LogoImg} {...props} />
);
