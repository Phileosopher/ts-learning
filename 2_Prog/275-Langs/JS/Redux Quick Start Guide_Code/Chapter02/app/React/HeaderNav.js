import React from "react";
import Link from "react-router-dom/Link";

import Header from "./Header";
import SocialMediaLinks from "./SocialMediaLinks";
import Logo from "./Logo";

const HeaderNav = () => (
  <div>
    <Logo />
    <Header />
    <SocialMediaLinks />
  </div>
);

export default HeaderNav;
