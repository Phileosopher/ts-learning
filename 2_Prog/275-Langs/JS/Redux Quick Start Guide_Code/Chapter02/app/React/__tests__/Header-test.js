import React from "react";
import renderer from "react-test-renderer";
import Header from "../Header";

jest.mock("react-router-dom/Link", () => "Link");

it("should render correctly", () => {
  const component = renderer.create(<Header />);
  expect(component.toJSON()).toMatchSnapshot();
});
