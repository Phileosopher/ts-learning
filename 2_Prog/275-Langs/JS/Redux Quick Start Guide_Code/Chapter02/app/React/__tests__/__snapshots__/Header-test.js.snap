// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<ul>
  <li>
    <Link
      to="/"
    >
      Home
    </Link>
  </li>
  <li>
    <Link
      to="/about"
    >
      Services
    </Link>
  </li>
  <li>
    <Link
      to="/topics"
    >
      Contact Us
    </Link>
  </li>
  <li>
    <Link
      to="/topics"
    >
      Login
    </Link>
  </li>
</ul>
`;
